compiler/rustc_mir_dataflow/src/value_analysis.rs

```diff
@@ -87,12 +87,16 @@ pub trait ValueAnalysis<'tcx> {
             StatementKind::CopyNonOverlapping(..) => {
                 // FIXME: What to do here?
             }
-            StatementKind::StorageLive(..)
-            | StatementKind::StorageDead(..)
-            | StatementKind::Deinit(_) => {
-                // Could perhaps use these.
+            StatementKind::StorageDead(local) => {
+                // It is UB to access an unallocated local.
+                state.flood(Place::from(*local).as_ref(), self.map());
+            }
+            StatementKind::Deinit(box place) => {
+                // It is UB to access `uninit` bytes.
```

Comment:
This isn't strictly speaking true. There are types (eg `MaybeUninit<Foo>`) at which it is DB to access uninit bytes. What makes this sound is that it is always sound to replace uninit bytes with whichever init bytes we like. This means that when such a read of the uninit bytes happens, we can pretend that we instead read *insert whichever bytes are convenient for us*.

Reply:
I agree that this should be worded differently. Also, it is probably related to https://github.com/rust-lang/rust/pull/101168#discussion_r962734777. The semantics here depend on what we decide there. Also, we surely need additional assumptions. For instance, constant propagation requires that unitialized integers (or at least BinOps with them) are UB (as far as I'm aware, the rules are not finalized here, but this simple case is hopefully undisputed).